crates/sync/anvil/src/init.rs

```diff
@@ -18,13 +21,35 @@ async fn receiver() -> ! {
     let mut rx = ethui_broadcast::subscribe_internal().await;
 
     loop {
-        if let Ok(InternalMsg::ResetAnvilListener { chain_id, http, ws }) = rx.recv().await {
-            trace!("resetting anvil listener for chain_id {}", chain_id);
-            reset_listener(chain_id, http, ws).await
+        match rx.recv().await {
+            Ok(InternalMsg::NetworkAdded(chain_id))
+            | Ok(InternalMsg::NetworkUpdated(chain_id))
+            | Ok(InternalMsg::NetworkRemoved(chain_id)) => {
+                if let Ok(Some((http, ws))) = anvil_network(chain_id).await {
+                    trace!("resetting anvil listener for chain_id {}", chain_id);
+                    reset_listener(chain_id, http, ws).await
+                }
+            }
+            _ => (),
         }
     }
 }
 
+async fn anvil_network(chain_id: u32) -> Result<Option<(Url, Url)>> {
+    let networks = Networks::read().await;
+
+    if let Some(network) = networks.get_network(chain_id) {
+        if networks.validate_chain_id(chain_id) && network.is_dev().await {
+            let http = Url::parse(&network.http_url)?;
+            let ws = Url::parse(&network.ws_url())?;
+
+            return Ok(Some((http, ws)));
+        }
+    }
```

Comment:
I don't like these nested ifs. can we move these to the same syntaax we used earlier today?